src/libraries/System.IO.Compression/tests/CompressionStreamUnitTests.ZLib.cs

```diff
@@ -19,44 +21,134 @@ public class ZLibStreamUnitTests : CompressionStreamUnitTestBase
 
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/47563")]
-        [Fact]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public void StreamCorruption_IsDetected()
         {
-            byte[] source = Enumerable.Range(0, 64).Select(i => (byte)i).ToArray();
-            var buffer = new byte[64];
-            byte[] compressedData;
-            using (var compressed = new MemoryStream())
-            using (Stream compressor = CreateStream(compressed, CompressionMode.Compress))
+            RemoteExecutor.Invoke(() =>
             {
-                foreach (byte b in source)
+                AppContext.SetSwitch("System.IO.Compression.UseStrictValidation", true);
+
+                byte[] source = Enumerable.Range(0, 64).Select(i => (byte)i).ToArray();
+                var buffer = new byte[64];
+                byte[] compressedData;
+                using (var compressed = new MemoryStream())
+                using (Stream compressor = CreateStream(compressed, CompressionMode.Compress))
                 {
-                    compressor.WriteByte(b);
+                    foreach (byte b in source)
+                    {
+                        compressor.WriteByte(b);
+                    }
+
+                    compressor.Dispose();
+                    compressedData = compressed.ToArray();
                 }
 
-                compressor.Dispose();
-                compressedData = compressed.ToArray();
-            }
+                for (int byteToCorrupt = 0; byteToCorrupt < compressedData.Length; byteToCorrupt++)
+                {
+                    // corrupt the data
+                    compressedData[byteToCorrupt]++;
 
-            for (int byteToCorrupt = 0; byteToCorrupt < compressedData.Length; byteToCorrupt++)
+                    using (var decompressedStream = new MemoryStream(compressedData))
+                    {
+                        using (Stream decompressor = CreateStream(decompressedStream, CompressionMode.Decompress))
+                        {
+                            Assert.Throws<InvalidDataException>(() =>
+                            {
+                                while (ZipFileTestBase.ReadAllBytes(decompressor, buffer, 0, buffer.Length) != 0);
+                            });
+                        }
+                    }
+
+                    // restore the data
+                    compressedData[byteToCorrupt]--;
+                }
+            }).Dispose();
+        }
+
+        [InlineData(TestScenario.ReadAsync)]
+        [InlineData(TestScenario.Read)]
+        [InlineData(TestScenario.Copy)]
+        [InlineData(TestScenario.CopyAsync)]
+        [InlineData(TestScenario.ReadByte)]
+        [InlineData(TestScenario.ReadByteAsync)]
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void StreamTruncation_IsDetected(TestScenario testScenario)
```

Comment:
This test is duplicated for Deflate, Gzip and Zlib tests. Why did you move it out of `CompressionStreamUnitTestBase`?

Reply:
https://github.com/dotnet/runtime/pull/75671/commits/fa4b613a9c3fe4f13d8b2fc147676ca2d2e92236

> abstracted classes are not supported by RemoteExecutor